SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt

```diff
@@ -0,0 +1,135 @@
+package app.cybrid.sdkandroid.components.bankAccounts.view
+
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import app.cybrid.cybrid_api_bank.client.apis.ExternalBankAccountsApi
+import app.cybrid.cybrid_api_bank.client.apis.WorkflowsApi
+import app.cybrid.cybrid_api_bank.client.infrastructure.ApiClient
+import app.cybrid.cybrid_api_bank.client.models.PostExternalBankAccountBankModel
+import app.cybrid.cybrid_api_bank.client.models.PostWorkflowBankModel
+import app.cybrid.cybrid_api_bank.client.models.WorkflowWithDetailsBankModel
+import app.cybrid.sdkandroid.AppModule
+import app.cybrid.sdkandroid.Cybrid
+import app.cybrid.sdkandroid.util.*
+import app.cybrid.sdkandroid.components.BankAccountsView.BankAccountsViewState as BankAccountsViewState
+import com.plaid.link.result.LinkAccount
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+
+class BankAccountsViewModel: ViewModel() {
+
+    private var workflowService = AppModule.getClient().createService(WorkflowsApi::class.java)
+    private var externalBankAccountsService = AppModule.getClient().createService(ExternalBankAccountsApi::class.java)
+
+    var customerGuid = Cybrid.instance.customerGuid
+
+    var UIState: MutableState<BankAccountsViewState> = mutableStateOf(BankAccountsViewState.LOADING)
+    var workflowJob: Polling? = null
+
+    var latestWorflow: WorkflowWithDetailsBankModel? = null
+
+    fun setDataProvider(dataProvider: ApiClient)  {
+
+        workflowService = dataProvider.createService(WorkflowsApi::class.java)
+        externalBankAccountsService = dataProvider.createService(ExternalBankAccountsApi::class.java)
+    }
+
+    suspend fun createWorkflow() {
+
+        Cybrid.instance.let { cybrid ->
+            if (!cybrid.invalidToken) {
+                viewModelScope.let { scope ->
+                    val waitFor = scope.async {
+
+                        val workflowResult = getResult {
+                            workflowService.createWorkflow(
+                                postWorkflowBankModel = PostWorkflowBankModel(
+                                    type = PostWorkflowBankModel.Type.plaid,
+                                    kind = PostWorkflowBankModel.Kind.create,
+                                    customerGuid = customerGuid,
+                                    externalBankAccountGuid = "3aa2bc487ae469bbc03cb42cf72e3aa3",
+                                    language = PostWorkflowBankModel.Language.en,
+                                    linkCustomizationName = "default",
+                                    androidPackageName = "app.cybrid.sdkandroid"
```

Comment:
These two values should defined in a constants file.

Reply:
Done!